### ISSUE TITLE
Feature Request: File Size Limit Parameter

### ISSUE THREAD

**Issue by alex-ander-prgrm** [darkdragn/party#36]
Not sure if you're taking requests or not, but could a file size parameter be added?

Some files on the party sites are massive, and I sometimes only need files up to a certain size.

**Comment by darkdragn** [comment]
Not a problem. I'm planning on updating some schemas this evening for the new thumbnails and some attachment attributes for t eh great gumroad purge.

I'll see what I can do about adding it. I'll probably add it as a skipped status and add details to the debug log.

**Comment by darkdragn** [comment]
Sorry about the delay, man. I got really distracted, my son needed open heart surgery in April so I kind of forgot the rest of the world existed. He's a toddler, so you know how the dad thing goes. 

I'll try getting around to this in the next couple of days. 

Again, I feel terrible about forgetting to do this and I'm sorry, man.

**Comment by darkdragn** [comment]
I just finished a proof of concept test:

```bash
-> % party kemono fanbox 3316400 --file-format "{ref.post_id}_{ref.index}_{ref.filename}" --sluglify --size-limit $((50*1024*1024))
Downloading from user: MだSたろう
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2364/2364 [07:43<00:00,  5.10it/s]
2024-10-01 22:49:09.348 | INFO     | party.cli:pull_user:218 - Output status: Counter({<StatusEnum.SUCCESS: 1>: 2010, <StatusEnum.TOO_LARGE: 8>: 287, <StatusEnum.DUPLICATE: 7>: 67})  
```

In that the file size limit is 50Mb. So, this is the point where it becomes a choose your own adventure. 

1. I can adjust the input size to scale to Mb if you'd prefer instead of the current bytes conversion (I used bash multiplication in that example to achieve that)
2. I currently don't have it on the .info, so update will ignore it. I can add it to .info, but everything is going to require an info update... or I can maybe come up with some additive run so you just need a new pull for each maintained folder followed by update working.
3. Currently it's only on pull, not on search or update. I can add it pretty easily and have it out in a version within the hour. 

Let me know what you think about these changes and I'll publish an update.

**Comment by darkdragn** [comment]
Rgr, rgr. Throwing the release together now. I just got off of work and I have some down time before my oldest gets home from school. (Wife's busy with the toddler)

I should have it out within the hour. I'll give you an update when it's done!

**Comment by darkdragn** [comment]
Done! Published in release v0.7.6

https://github.com/darkdragn/party/releases/tag/v0.7.6
